Modix.Bot/Behaviors/PromotionLoggingHandler.cs

```diff
@@ -37,39 +41,127 @@ public Task Handle(PromotionActionCreated notification, CancellationToken cancel
             {
-                var promotionAction = await PromotionsService.GetPromotionActionSummaryAsync(promotionActionId);
+                var message = await FormatPromotionLogEntry(promotionActionId, data);
 
-                if (!_renderTemplates.TryGetValue((promotionAction.Type, promotionAction.NewComment?.Sentiment, promotionAction.Campaign?.Outcome), out var renderTemplate))
+                if (message == null)
                     return;
 
-                var message = string.Format(renderTemplate,
-                    promotionAction.Created.UtcDateTime.ToString("HH:mm:ss"),
-                    promotionAction.Campaign?.Id,
-                    promotionAction.Campaign?.Subject.DisplayName,
-                    promotionAction.Campaign?.Subject.Id,
-                    promotionAction.Campaign?.TargetRole.Name,
-                    promotionAction.Campaign?.TargetRole.Id,
-                    promotionAction.NewComment?.Campaign.Id,
-                    promotionAction.NewComment?.Campaign.Subject.DisplayName,
-                    promotionAction.NewComment?.Campaign.Subject.Id,
-                    promotionAction.NewComment?.Campaign.TargetRole.Name,
-                    promotionAction.NewComment?.Campaign.TargetRole.Id,
-                    promotionAction.NewComment?.Content);
-
                 await DesignatedChannelService.SendToDesignatedChannelsAsync(
                     await DiscordClient.GetGuildAsync(data.GuildId), DesignatedChannelType.PromotionLog, message);
             }
-            catch (Exception ex)
+
+            if (await DesignatedChannelService.AnyDesignatedChannelAsync(data.GuildId, DesignatedChannelType.PromotionNotifications))
             {
-                var text = Newtonsoft.Json.JsonConvert.SerializeObject(ex);
+                var embed = await FormatPromotionNotification(promotionActionId, data);
+
+                if (embed == null)
+                    return;
+
+                await DesignatedChannelService.SendToDesignatedChannelsAsync(
+                    await DiscordClient.GetGuildAsync(data.GuildId), DesignatedChannelType.PromotionNotifications, "", embed);
             }
         }
 
+        private async Task<Embed> FormatPromotionNotification(long promotionActionId, PromotionActionCreationData data)
+        {
+            var promotionAction = await PromotionsService.GetPromotionActionSummaryAsync(promotionActionId);
+            var targetCampaign = promotionAction.Campaign ?? promotionAction.NewComment.Campaign;
+
+            var embed = new EmbedBuilder();
+
+            //Because we have comment creation as a separate operation from starting the campaign,
+            //we don't have access to the "initial" comment when a campaign is created. So, we have to
+            //note that a campaign was created, and actually send the log message when the first comment
+            //is created (containing the comment body)
+            switch (promotionAction.Type)
```

Comment:
A rather clever approach to the problem. We can probably think about modifying the promotion creation logic to have the repository create both the comment and the campaign, so at least they're guarantee to both be present at the same time. Or maybe we really should just merge the initial comments up into the campaign itself, since we now have several special rules for "initial" comments versus "non-initial".

Alternatively, the easy way to tell if a comment is "initial" or not is to see if the comment and campaign were created by the same user. Maybe we could add this as a virtual/calculated column on `PromotionCommentBrief` and `PromotionCommentSummary`.

Reply:
Doesn't make sense to skip rejections if we still display creations. If someone sees that they were nominated, and then never approved, it amounts to the same thing.

Reply:
How about we approach it the other way then - I now have it embed the first 5 approving (for accepted) or first 5 opposing (for rejected) comments as well.

Reply:
The differentiation is this - "PromotionNotifications" is public-facing channel, "PromotionLog" is a raw log for staff. 

With my most recent change, comments are only shown within the "campaign closed" notification, not as they come in - so if there's a few negative comments, but they're deemed invalid and the promotion goes through anyway, they won't be part of the notification. Similarly, if the campaign is rejected, the user still gets to see some of the comments and get an idea of why.

A while back I also considered the idea of having "rejection reasons" for staff to type in, but that's somewhat outside the scope of this PR - plus, most of the time the comments are self-explanatory.

Reply:
We're all on the same page, I think, it's just a question of WHAT we want to actually make public. I personally think the only thing we wanna make public is when people are promoted, cause I think anything more than that will either piss people off, or will be detailed enough that we can just make the whole damn feature public.

If we're at the point of having the public-facing channel include chunks of comment data as well, we might as well just make the whole log channel public, and if that means making it prettier at the same time, well that was always the intention.

Also, yeah, it sounds more and more to me like we should have "reasons" be included directly on the campaigns, both for creation and rejection. We can definitely consider that as a separate issue.